code/modules/supermatter/supermatter.dm

```diff
@@ -278,6 +279,10 @@
 		if (!(l in oview(rad_range, src)) && !(l in range(src, round(rad_range * 2/3))))
 			continue
 		l.apply_effect(rads, IRRADIATE, blocked = l.getarmor(null, "rad"))
+		if(l.is_diona())
+			l.adjustToxLoss(-rads)
+			if(last_taste_time + 800 < world.time) // Not to spam message
```

Comment:
You're not setting `last_taste_time = world.time` in this chain. So the cool down won't work. Toss a boolean at the loop and set the cooldown _after_ the loop. Otherwise only one diona in range will get the memo.